### ISSUE TITLE
False-Positive: pkg:composer/nette/schema@v1.2.5

### ISSUE THREAD

**Issue by ejunker** [owasp-dep-scan/dep-scan#215]
### PURL of wrongly matched component

pkg:composer/nette/schema@v1.2.5

### Depscan findings

```
██████╗ ███████╗██████╗ ███████╗ ██████╗ █████╗ ███╗   ██╗
██╔══██╗██╔════╝██╔══██╗██╔════╝██╔════╝██╔══██╗████╗  ██║
██║  ██║█████╗  ██████╔╝███████╗██║     ███████║██╔██╗ ██║
██║  ██║██╔══╝  ██╔═══╝ ╚════██║██║     ██╔══██║██║╚██╗██║
██████╔╝███████╗██║     ███████║╚██████╗██║  ██║██║ ╚████║
╚═════╝ ╚══════╝╚═╝     ╚══════╝ ╚═════╝╚═╝  ╚═╝╚═╝  ╚═══╝

INFO [2024-01-22 22:55:46,569] Performing regular scan for None using plugin composer

                       Dependency Scan Results (COMPOSER)
╔═════════════════════════════════╤═══════════╤═════════════╤══════════╤═══════╗
║ CVE                             │ Insights  │ Fix Version │ Severity │ Score ║
╟─────────────────────────────────┼───────────┼─────────────┼──────────┼───────╢
║ schema@v1.2.5 ⬅ CVE-2022-33154  │           │ 1.13.1      │ MEDIUM   │   5.4 ║
╚═════════════════════════════════╧═══════════╧═════════════╧══════════╧═══════╝
╭────────────── Recommendation ───────────────╮
│ ✅ No package requires immediate attention. │
╰─────────────────────────────────────────────╯
```

**Comment by prabhu** [comment]
Thank you. This and #214 are due to the use of the `_project` alias during the search.

https://github.com/owasp-dep-scan/dep-scan/blob/master/depscan/lib/normalize.py#L105

I believe this is not required here since there is already a vendor `nette` in the input.

Could you send a PR by testing with the code snippet below?

https://github.com/owasp-dep-scan/dep-scan/blob/master/depscan/lib/normalize.py#L103

```
if not vendor and purl.startswith("pkg:composer"):
```

**Comment by ejunker** [comment]
I'm using Docker to run dep-scan so I don't know how to easily try your changes. If it helps here is a minimal composer.lock file that you can test with.

<details>
<summary>composer.lock</summary>

```json
{
    "_readme": [
        "This file locks the dependencies of your project to a known state",
        "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
        "This file is @generated automatically"
    ],
    "content-hash": "27d8dfa70d1d19eca249e326b18f2f3f",
    "packages": [
        {
            "name": "nette/schema",
            "version": "v1.3.0",
            "source": {
                "type": "git",
                "url": "https://github.com/nette/schema.git",
                "reference": "a6d3a6d1f545f01ef38e60f375d1cf1f4de98188"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/nette/schema/zipball/a6d3a6d1f545f01ef38e60f375d1cf1f4de98188",
                "reference": "a6d3a6d1f545f01ef38e60f375d1cf1f4de98188",
                "shasum": ""
            },
            "require": {
                "nette/utils": "^4.0",
                "php": "8.1 - 8.3"
            },
            "require-dev": {
                "nette/tester": "^2.4",
                "phpstan/phpstan-nette": "^1.0",
                "tracy/tracy": "^2.8"
            },
            "type": "library",
            "extra": {
                "branch-alias": {
                    "dev-master": "1.3-dev"
                }
            },
            "autoload": {
                "classmap": [
                    "src/"
                ]
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "BSD-3-Clause",
                "GPL-2.0-only",
                "GPL-3.0-only"
            ],
            "authors": [
                {
                    "name": "David Grudl",
                    "homepage": "https://davidgrudl.com"
                },
                {
                    "name": "Nette Community",
                    "homepage": "https://nette.org/contributors"
                }
            ],
            "description": "📐 Nette Schema: validating data structures against a given Schema.",
            "homepage": "https://nette.org",
            "keywords": [
                "config",
                "nette"
            ],
            "support": {
                "issues": "https://github.com/nette/schema/issues",
                "source": "https://github.com/nette/schema/tree/v1.3.0"
            },
            "time": "2023-12-11T11:54:22+00:00"
        },
        {
            "name": "nette/utils",
            "version": "v4.0.4",
            "source": {
                "type": "git",
                "url": "https://github.com/nette/utils.git",
                "reference": "d3ad0aa3b9f934602cb3e3902ebccf10be34d218"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/nette/utils/zipball/d3ad0aa3b9f934602cb3e3902ebccf10be34d218",
                "reference": "d3ad0aa3b9f934602cb3e3902ebccf10be34d218",
                "shasum": ""
            },
            "require": {
                "php": ">=8.0 <8.4"
            },
            "conflict": {
                "nette/finder": "<3",
                "nette/schema": "<1.2.2"
            },
            "require-dev": {
                "jetbrains/phpstorm-attributes": "dev-master",
                "nette/tester": "^2.5",
                "phpstan/phpstan": "^1.0",
                "tracy/tracy": "^2.9"
            },
            "suggest": {
                "ext-gd": "to use Image",
                "ext-iconv": "to use Strings::webalize(), toAscii(), chr() and reverse()",
                "ext-intl": "to use Strings::webalize(), toAscii(), normalize() and compare()",
                "ext-json": "to use Nette\\Utils\\Json",
                "ext-mbstring": "to use Strings::lower() etc...",
                "ext-tokenizer": "to use Nette\\Utils\\Reflection::getUseStatements()"
            },
            "type": "library",
            "extra": {
                "branch-alias": {
                    "dev-master": "4.0-dev"
                }
            },
            "autoload": {
                "classmap": [
                    "src/"
                ]
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "BSD-3-Clause",
                "GPL-2.0-only",
                "GPL-3.0-only"
            ],
            "authors": [
                {
                    "name": "David Grudl",
                    "homepage": "https://davidgrudl.com"
                },
                {
                    "name": "Nette Community",
                    "homepage": "https://nette.org/contributors"
                }
            ],
            "description": "🛠  Nette Utils: lightweight utilities for string & array manipulation, image handling, safe JSON encoding/decoding, validation, slug or strong password generating etc.",
            "homepage": "https://nette.org",
            "keywords": [
                "array",
                "core",
                "datetime",
                "images",
                "json",
                "nette",
                "paginator",
                "password",
                "slugify",
                "string",
                "unicode",
                "utf-8",
                "utility",
                "validation"
            ],
            "support": {
                "issues": "https://github.com/nette/utils/issues",
                "source": "https://github.com/nette/utils/tree/v4.0.4"
            },
            "time": "2024-01-17T16:50:36+00:00"
        }
    ],
    "packages-dev": [],
    "aliases": [],
    "minimum-stability": "stable",
    "stability-flags": [],
    "prefer-stable": false,
    "prefer-lowest": false,
    "platform": [],
    "platform-dev": [],
    "plugin-api-version": "2.6.0"
}
```
</details>

**Comment by prabhu** [comment]
```python depscan/cli.py --purl "pkg:composer/nette/schema@v1.2.5" --reports-dir /tmp/reports                                                           ok  depscan py 

██████╗ ███████╗██████╗ ███████╗ ██████╗ █████╗ ███╗   ██╗
██╔══██╗██╔════╝██╔══██╗██╔════╝██╔════╝██╔══██╗████╗  ██║
██║  ██║█████╗  ██████╔╝███████╗██║     ███████║██╔██╗ ██║
██║  ██║██╔══╝  ██╔═══╝ ╚════██║██║     ██╔══██║██║╚██╗██║
██████╔╝███████╗██║     ███████║╚██████╗██║  ██║██║ ╚████║
╚═════╝ ╚══════╝╚═╝     ╚══════╝ ╚═════╝╚═╝  ╚═╝╚═╝  ╚═══╝

INFO [2024-01-22 23:29:53,573] Performing regular scan for None using plugin composer
INFO [2024-01-22 23:29:53,573] No oss vulnerabilities detected ✅```

**Comment by prabhu** [comment]
Fixed in 5.2.1
